src/views/Shift.vue

```diff
@@ -52,11 +52,20 @@ export default {
         this.nextOrder();
       }
       if (this.order.hasProducts()) {
+        this.orderBeingUpdated = true;
         await salesService.updateOrder(this.order, parseInt(this.shiftId)).then((order) => (this.order = order));
+        this.orderBeingUpdated = false;
```

Comment:
```suggestion
        clearTimeout(this.fetchOrder);
        this.order = await salesService.updateOrder(this.order, parseInt(this.shiftId));
        this.fetchOrder = setTimeout(this.fetchOrderUpdates, this.ORDER_REFRESH_RATE);
```

I don't understand why the timeout is not just cleared here instead of setting a variable, this would simplify the contents of fetchOrderUpdates as well.